engine/lite/kafka/runtime/src/main/scala/pl/touk/nussknacker/engine/lite/kafka/LoopUntilClosed.scala

```diff
@@ -19,8 +19,8 @@ class TaskRunner(taskName: String,
                  singleRun: String => Task,
                  terminationTimeout: Duration,
                  waitAfterFailureDelay: FiniteDuration) extends AutoCloseable with LazyLogging {
-  def status(): TaskStatus = tasks.find(p => p.status == Restarting).map(_.status
-  ).getOrElse(Running)
+  def status(): TaskStatus = tasks
```

Comment:
I would be for precedence (from most important to lower)
- restarting
- during_deploy
- running
IMO restarting is more important than during deploy because it is some failure situation